pkg/util/iptables/iptables.go

```diff
@@ -120,6 +120,56 @@ const MinCheckVersion = "1.4.11"
 const MinWaitVersion = "1.4.20"
 const MinWait2Version = "1.4.22"
 
+// no-op implemenatation of Interface
+type fake struct{}
+
+func NewFake() *fake {
```

Comment:
This whole file seems to no longer be part of the PR?

As you said: putting it in a fake_ file helps with readability but not linking.

I don't understand your two options.  If you put it in a test sub-pkg, you can still implement the Interface, it's juts that the caller calls iptables_testing.NewFake() intead of iptables.NewFake()